manager/controllers/utils/utils.go

```diff
@@ -145,3 +151,45 @@ func GetFybrikApplicationUUIDfromAnnotations(annotations map[string]string) stri
 	}
 	return uuid
 }
+
+// UpdateFinalizers adds or removes finalizers for a resource
+func UpdateFinalizers(ctx context.Context, cl client.Client, obj client.Object) error {
+	finalizers := obj.GetFinalizers()
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Retrieve the latest version of the object before attempting update
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+		if err := cl.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
+			return err
+		}
+		obj.SetFinalizers(finalizers)
+		return cl.Update(ctx, obj)
+	})
+}
```

Comment:
you have the object, why not to use it first 
```
func UpdateFinalizers(ctx context.Context, cl client.Client, obj client.Object) error {
	finalizers := obj.GetFinalizers()
	if err := cl.Update(ctx, obj); err != nil {
		if ! errors.IsConflict(err) {
			return err
		}
		return retry.RetryOnConflict(retry.DefaultRetry, func() error {
			// Retrieve the latest version of the object before attempting update
			// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
			if err := cl.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
				return err
			}
			obj.SetFinalizers(finalizers)
			return cl.Update(ctx, obj)
		})
	}
	return nil
}
```

Reply:
Fixed